vllm/distributed/object_list_ops.py

```diff
@@ -0,0 +1,123 @@
+"""
+This file is necessary until new version of torch.distributed is released with
+https://github.com/pytorch/pytorch/commit/b96b1e8cff029bb0a73283e6e7f6cc240313f1dc
+"""
+import torch
+import torch.distributed as dist
+from torch.distributed.distributed_c10d import (_get_pg_default_device,
+                                                _object_to_tensor,
+                                                _tensor_to_object)
+
+
+def send_object_list(object_list, dst, group=None, device=None):
```

Comment:
to be honest, we never send object list literally. all we need is just `send_object`. can we add `send_object` to group coordinator, with the signature as `broadcast_object`? in the implementation, we can first send a int tensor with size 1 to send the size info, and then send serialized python object, finally send all tensors.

Reply:
Hey @youkaichao it doesn't look like we actually have `broadcast_object` in group coordinator. We only support `broadcast_object_list`.

I think I kind of understand what you are saying, but will there be a big difference? I just tried to make it completely symmetric with `broadcast*` ops to avoid spending more time on it. Trying to see if I'm missing something

Reply:
it's here https://github.com/vllm-project/vllm/pull/5399 . too many things recently, that i didn't find time to finish it :(

Reply:
Ohhh got it - in that case can we keep doing `send_object` as part of that PR or a separate one? Just want to see if we can keep this PR simple as it's just symmetric with the current state of `broadcast*` for now.

Reply:
I prefer we only keep `send_object` in this PR, in the future i want to deprecate `broadcast_object_list`, because it is confusing. `list` is also an object, right?

Reply:
Got it. To be clear, you are not referring to using `shm` for this right? I think because #5399 does not do it this way:
>  in the implementation, we can first send a int tensor with size 1 to send the size info, and then send serialized python object, finally send all tensors.

It either uses `shm` or uses `broadcast_object_list` as a fallback

Reply:
yeah, you don't need to use `shm` for this.

Reply:
Made the change as per your comments